sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/Util.java

```diff
@@ -371,9 +371,11 @@ public static URI convertNamespaceToEndPointURI(String namespaceName) {
     }
 
     public static ClientSettings getClientSettingsFromConnectionStringBuilder(ConnectionStringBuilder builder) {
-        TokenProvider tokenProvider;
-        if (builder.getAuthentication() != null && builder.getAuthentication().equalsIgnoreCase(ConnectionStringBuilder.MANAGED_IDENTITY_AUTHENTICATION)) {
-            tokenProvider = new ManagedIdentityTokenProvider();
+        TokenProvider tokenProvider = null;
+        if (builder.getAuthentication() != null) {
+            if (builder.getAuthentication().replace(" ", "").equalsIgnoreCase(ConnectionStringBuilder.MANAGED_IDENTITY_AUTHENTICATION.replace(" ", ""))) {
```

Comment:
>.replace(" ", "") [](start = 133, length = 17)

Why do you need this? That's a constant defined in the code. Why are you trying to remove spaces? Just remove spaces, if any, from the declared constant value.